branca/element.py

```diff
@@ -88,6 +89,14 @@ def __setstate__(self, state: dict):
 
         self.__dict__.update(state)
 
+    def clone(self):
+        """creates a new copy of an element, but with a unique identifier
+        and without the prior version's relationship to a parent object"""
+        clone = copy(self)
```

Comment:
I wonder about cases where this copy is not sufficient. 

- What if the element has children?
- Does it matter if we don't create a new Template object?
- Folium has all kinds of classes, maybe there are some that have class attributes for which a simple copy is not sufficient?

I'm not sure how we can know this clone method works robustly for all Folium classes.

And if we use deepcopy, is that too much?